rust/connlib/tunnel/src/client.rs

```diff
@@ -1779,7 +1782,7 @@ impl ClientState {
         }
     }
 
-    fn update_dns_mapping(&mut self) {
+    fn update_dns_mapping(&mut self, new_search_domain: Option<DomainName>) {
```

Comment:
The search domain has nothing to do with the DNS mapping so this shouldn't be in here ...

I'll see if I can fix this up.